### PR TITLE
feat: [CDS-73647]: Expression Playground phase 1 

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Compiled View support in the right drawer variables panel 

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/5aaa5132-adc6-402b-b880-e3d85ad374d4

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/components/CopyToClipBoard/CopyToClipBoard.tsx [hunk 1]
```diff
@@ -17,11 +17,13 @@ interface CopyToClipboardProps {
   content: string
   showFeedback?: boolean
   iconSize?: number
+  showPopover?: boolean
```
Comment (PratyushGar) [inline src/modules/10-common/components/CopyToClipBoard/CopyToClipBoard.tsx:20]: You can rename this to `hidePopover` as this is an optional prop

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.tsx [hunk 2]
```diff
@@ -138,13 +141,28 @@ export function PipelineVariablesWithRef(
 
   const readOnly = isReadonly || !!pipeline.template
 
+  const handleCompiledModeChange = (checked: boolean): void => {
+    if (checked) {
+      setIsCompiledMode?.(true)
+    } else {
+      setIsCompiledMode?.(false)
+    }
+  }
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.tsx:150]:
```suggestion
const handleCompiledModeChange = (checked: boolean): void => {
  setIsCompiledMode?.(checked);
}
```

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/Cards/PipelineCard.tsx [hunk 3]
```diff
@@ -36,6 +37,8 @@ export interface PipelineCardProps {
   updatePipeline(pipeline: PipelineInfoConfig): void
   readonly?: boolean
   allowableTypes: AllowedTypes
+  isCompiledMode?: boolean
+  compiledModeMetadataMap?: any
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/Cards/PipelineCard.tsx:41]:
```suggestion
  compiledModeMetadataMap?: MetadataMapObject
```

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    <Layout.Horizontal className={css.compiledModeHeader}>
+      <Toggle label="View in Compiled Mode" onToggle={handleCompiledModeChange} checked={isCompiledMode} />
+      {isCompiledMode && (
+        <FormInput.Select
+          items={executionIdSelectOptions}
+          onChange={handleExecutionIdChange}
+          name="executionId"
+          placeholder="ExecutionID"
+          value={{ label: executionId as string, value: executionId as string }}
+          style={{ marginBottom: '0px', paddingLeft: '10px' }}
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:91]: Move this inline style to class

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    requestOptions: { headers: { 'content-type': 'application/json' } }
+  })
+
+  React.useEffect(() => {
+    setCompiledModeMetadataMap?.(expressionEvaluatedResponse?.data?.mapExpression as MetadataMapObject)
+  }, [expressionEvaluatedResponse, setCompiledModeMetadataMap])
+
+  return (
+    <Layout.Horizontal className={css.compiledModeHeader}>
+      <Toggle label="View in Compiled Mode" onToggle={handleCompiledModeChange} checked={isCompiledMode} />
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:83]: `View in Compiled Mode` Move this to YAML

File: src/modules/10-common/components/TextWithExpressions/TextWithExpression.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      // Word matches the regex, render the Popover component
+      return <ResolvedValueComponent metadataMap={metadataMap} word={word} fqnPath={fqnPath} />
+    } else {
+      // Word doesn't match the regex, render it as is
+      return `${word} `
+    }
+  })
+
+  return (
+    <span className={cx(customClassName, css.expressionWrapper)} style={{ display: 'flex' }}>
```
Comment (PratyushGar) [inline src/modules/10-common/components/TextWithExpressions/TextWithExpression.tsx:123]: `<Text inline >` can be used here

File: src/modules/10-common/components/CopyToClipBoard/CopyToClipBoard.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/10-common/components/CopyToClipBoard/CopyToClipBoard.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/Cards/PipelineCard.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/Cards/PipelineCard.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 5 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: done

File: src/modules/10-common/components/TextWithExpressions/TextWithExpression.tsx (hunk 6 above)
Comment (vikrant-harness) [reply src/modules/10-common/components/TextWithExpressions/TextWithExpression.tsx]: Done

**Review by PratyushGar — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Comment by vikrant-harness** [comment]
trigger featurenamecheck

**Pull request merged**